arho_feature_template/project/layers/__init__.py

```diff
@@ -109,3 +109,11 @@ def get_attribute_by_id(cls, target_attribute: str, id_: str) -> Any | None:
     def get_id_by_attribute(cls, attribute: str, attribute_value: str) -> str | None:
         id_ = cls.get_attribute_value_by_another_attribute_value("id", attribute, attribute_value)
         return cast(str, id_) if id_ else id_
+
+    @classmethod
+    def get_features_by_ids(cls, ids: list[str], field_name: str) -> list[QgsFeature] | None:
```

Comment:
`in()` hakutoiminnallisuuden voisi toteuttaa sisään noihin nykyisiin hakumetodeihin esim. `get_features_by_attribute_value`, `get_attribute_values_by_another_attribute_value`. Ne voisi tarkastaa, että jos filteröitävä arvo on `str` käytetään `=` operaattoria, jos `list|tuple|set` niin `in ()` operaattoria ja jos None niin `IS NULL`.

Oikeastaan kannattaa varmaan toteuttaa metodi `create_filter_expression(attribute, value) -> str` joka tuon tekee.